network_wrangler/roadway/network.py

```diff
@@ -611,6 +646,242 @@ def delete_nodes(
             self.nodes_df, del_node_ids, ignore_missing=selection.ignore_missing
         )
 
+    def split_link(  # noqa: PLR0912, PLR0915
+        self,
+        A: int,
```

Comment:
Should this be a [seleciton dictionary](https://github.com/network-wrangler/network_wrangler/blob/main/network_wrangler/models/projects/roadway_selection.py#L49) and then just do the selection using those methods? I'm not wedded to that but am just trying to match existing patterns.  Happy either way.